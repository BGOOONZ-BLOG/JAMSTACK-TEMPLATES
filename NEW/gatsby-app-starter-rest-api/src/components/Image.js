import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import Img from 'gatsby-image'

export default () => {
  const { placeholderImage } = useStaticQuery(graphql`
    query {
      placeholderImage: file(relativePath: { eq: "gatsby-astronaut.png" }) {
        childImageSharp {
          fluid(maxWidth: 300) {
            ...GatsbyImageSharpFluid
          }
        }
      }
    }
  `)

  return <Img fluid={placeholderImage.childImageSharp.fluid} />
}
