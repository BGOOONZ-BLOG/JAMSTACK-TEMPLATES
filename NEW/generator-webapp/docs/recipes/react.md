# Setting up React and JSX

This recipe gets you set up with React, including precompilation of JSX into JavaScript, integrated with LiveReload.

## Steps

### 1. Add dependencies

Install [gulp-babel](https://github.com/babel/gulp-babel), and the required presets for transforming JSX templates into vanilla JavaScript:

```
$ npm install --save-dev gulp-babel babel-preset-es2015 babel-preset-react
```

Install [React](https://github.com/facebook/react) itself as a Bower component:

```
$ bower install --save react
```

Run the wiredep task to insert a script tag into your `app/index.html`:

```
$ gulp wiredep
```

### 2. Create a `templates` task

This task preprocesses `.jsx` files into pure JavaScript and outputs them in `.tmp/scripts`.

```js
gulp.task('templates', () => {
  return gulp.src('app/scripts/**/*.jsx')
    .pipe($.sourcemaps.init())
    .pipe($.babel({
      presets: ['es2015','react']
    }))
    .pipe($.sourcemaps.write())
    .pipe(gulp.dest('.tmp/scripts'));
});
```

### 3. Add `templates` as a dependency of `html` and `serve`

```js
gulp.task('html', ['styles', 'templates'], () => {
  ...
```

```js
gulp.task('serve', ['styles', 'templates', 'fonts'], () => {
  ...
```

* The `serve` dependency means the generated `.js` files will be ready in `.tmp/scripts` before the server starts up
* The `html` dependency means your JSX also gets compiled as part of the `gulp build` sequence – before the `html` task starts, so that the `.js` files are generated in time for [gulp-useref](https://github.com/jonkemp/gulp-useref) to concatenate them.

### 4. Edit your `serve` task

Edit your `serve` task so that (a) editing a `.jsx` file triggers the `templates` task, and (b) the browser is refreshed whenever a `.js` file is generated in `.tmp/scripts`:

```diff
 gulp.task('serve', ['styles', 'templates', 'fonts'], () => {
   ...
   gulp.watch([
     'app/*.html',
     '.tmp/styles/**/*.css',
     'app/scripts/**/*.js',
+    '.tmp/scripts/**/*.js',
     'app/images/**/*'
   ]).on('change', reload);

   gulp.watch('app/styles/**/*.scss', ['styles', reload]);
+  gulp.watch('app/scripts/**/*.jsx', ['templates', reload]);
   gulp.watch('bower.json', ['wiredep', 'fonts', reload]);
 });
```


## Usage

- Put your `.jsx` files anywhere in `app/scripts`, but include them in your HTML as if they're `.js` files – e.g. for `app/scripts/foo.jsx`, use `<script src="scripts/foo.js"></script>`.

- It's fine to have a mixture of `.js` and `.jsx` files in your `app/scripts`.
