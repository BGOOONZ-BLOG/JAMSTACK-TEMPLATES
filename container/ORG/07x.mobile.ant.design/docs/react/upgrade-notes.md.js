webpackJsonp([108],{884:function(e,t,l){"use strict";function n(e){return e&&e.__esModule?e:{"default":e}}var a=l(1),o=(n(a),l(4));n(o);e.exports={content:["article",["p","\u6b64\u5904\u7740\u91cd\u5217\u51fa\u5347\u7ea7\u4e2d\u7684\u4e0d\u517c\u5bb9\u53d8\u5316\u548c\u63a8\u8350\u6539\u52a8\u3002\u6240\u6709\u53d8\u52a8\u8bf7\u89c1 ",["a",{title:null,href:"/changelog"},"Changelog"],"\u3002"],["h2","0.5.x => 0.6.0"],["p","\u8bf7\u67e5\u770b ",["a",{title:null,href:"/changelog#0.6.0"},"Changelog"],"\u3002"]],meta:{order:3,english:"\u5347\u7ea7\u6307\u5357",filename:"docs/react/upgrade-notes.md"},toc:["ul",["li",["a",{href:"#0.5.x-=>-0.6.0"},"0.5.x => 0.6.0"]]]}}});