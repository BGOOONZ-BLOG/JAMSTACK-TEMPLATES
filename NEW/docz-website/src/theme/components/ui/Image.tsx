import * as React from 'react'
import { SFC, useState } from 'react'
import Lightbox from 'react-images'
import styled from 'styled-components'

const Wrapper = styled.div`
  position: relative;
  margin: 30px 0;
  border: 1px solid ${p => p.theme.colors.grayLight};
`

const ImageStyled = styled.img`
  display: block;
  width: 100%;
  padding: 5px;
`

type ImageProps = React.ImgHTMLAttributes<any>

export const Image: SFC<ImageProps> = props => {
  const [opened, setOpened] = useState(false)
  const toggle = () => setOpened(s => !s)

  return (
    <Wrapper>
      <ImageStyled {...props} />
      {opened && (
        <Lightbox
          images={[{ src: props.src }]}
          isOpen={opened}
          onClose={toggle}
        />
      )}
    </Wrapper>
  )
}
