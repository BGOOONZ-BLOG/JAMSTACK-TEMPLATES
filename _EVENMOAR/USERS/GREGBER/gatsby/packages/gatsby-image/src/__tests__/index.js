import React from "react"
import { render, cleanup, fireEvent } from "@testing-library/react"
import MatchMediaMock from "jest-matchmedia-mock"
import Image from "../"

afterAll(cleanup)

const fixedShapeMock = {
  width: 100,
  height: 100,
  src: `test_image.jpg`,
  srcSet: `some srcSet`,
  srcSetWebp: `some srcSetWebp`,
  base64: `string_of_base64`,
}

const fluidShapeMock = {
  aspectRatio: 1.5,
  src: `test_image.jpg`,
  srcSet: `some srcSet`,
  srcSetWebp: `some srcSetWebp`,
  sizes: `(max-width: 600px) 100vw, 600px`,
  base64: `string_of_base64`,
}

const fixedImagesShapeMock = [
  {
    width: 100,
    height: 100,
    src: `test_image.jpg`,
    srcSet: `some srcSet`,
    srcSetWebp: `some srcSetWebp`,
    base64: `string_of_base64`,
  },
  {
    width: 300,
    height: 300,
    src: `test_image_2.jpg`,
    srcSet: `some other srcSet`,
    srcSetWebp: `some other srcSetWebp`,
    base64: `other_string_of_base64`,
    media: `only screen and (min-width: 768px)`,
  },
]

const fluidImagesShapeMock = [
  {
    aspectRatio: 2,
    src: `test_image.jpg`,
    srcSet: `some srcSet`,
    srcSetWebp: `some srcSetWebp`,
    sizes: `(max-width: 600px) 100vw, 600px`,
    base64: `string_of_base64`,
  },
  {
    aspectRatio: 3,
    src: `test_image_2.jpg`,
    srcSet: `some other srcSet`,
    srcSetWebp: `some other srcSetWebp`,
    sizes: `(max-width: 600px) 100vw, 600px`,
    base64: `string_of_base64`,
    media: `only screen and (min-width: 768px)`,
  },
]

const setup = (
  fluid = false,
  props = {},
  onLoad = () => {},
  onError = () => {}
) => {
  const { container } = render(
    <Image
      backgroundColor
      className={`fixedImage`}
      style={{ display: `inline` }}
      title={`Title for the image`}
      alt={`Alt text for the image`}
      crossOrigin={`anonymous`}
      draggable={true}
      {...(fluid && { fluid: fluidShapeMock })}
      {...(!fluid && { fixed: fixedShapeMock })}
      onLoad={onLoad}
      onError={onError}
      itemProp={`item-prop-for-the-image`}
      placeholderStyle={{ color: `red` }}
      placeholderClassName={`placeholder`}
      {...props}
    />
  )

  return container
}

const setupImages = (
  fluidImages = false,
  onLoad = () => {},
  onError = () => {}
) => {
  const { container } = render(
    <Image
      backgroundColor
      className={`fixedImage`}
      style={{ display: `inline` }}
      title={`Title for the image`}
      alt={`Alt text for the image`}
      crossOrigin={`anonymous`}
      {...(fluidImages && { fluid: fluidImagesShapeMock })}
      {...(!fluidImages && { fixed: fixedImagesShapeMock })}
      onLoad={onLoad}
      onError={onError}
      itemProp={`item-prop-for-the-image`}
      placeholderStyle={{ color: `red` }}
      placeholderClassName={`placeholder`}
    />
  )

  return container
}

let matchMedia
describe(`<Image />`, () => {
  beforeAll(() => {
    matchMedia = new MatchMediaMock()
  })
  afterEach(() => {
    matchMedia.clear()
  })

  it(`should render fixed size images`, () => {
    const component = setup()
    expect(component).toMatchSnapshot()
  })

  it(`should render fluid images`, () => {
    const component = setup(true)
    expect(component).toMatchSnapshot()
  })

  it(`should render multiple fixed image variants`, () => {
    const component = setupImages()
    expect(component).toMatchSnapshot()
  })

  it(`should render multiple fluid image variants`, () => {
    const component = setupImages(true)
    expect(component).toMatchSnapshot()
  })

  it(`should have correct src, title, alt, and crossOrigin attributes`, () => {
    const imageTag = setup().querySelector(`picture img`)
    expect(imageTag.getAttribute(`src`)).toEqual(`test_image.jpg`)
    expect(imageTag.getAttribute(`srcSet`)).toEqual(`some srcSet`)
    expect(imageTag.getAttribute(`title`)).toEqual(`Title for the image`)
    expect(imageTag.getAttribute(`alt`)).toEqual(`Alt text for the image`)
    expect(imageTag.getAttribute(`crossOrigin`)).toEqual(`anonymous`)
    expect(imageTag.getAttribute(`loading`)).toEqual(`lazy`)
    expect(imageTag.getAttribute(`draggable`)).toEqual(`true`)
  })

  it(`should have correct placeholder src, title, style and class attributes`, () => {
    const placeholderImageTag = setup().querySelector(`img`)
    expect(placeholderImageTag.getAttribute(`src`)).toEqual(`string_of_base64`)
    expect(placeholderImageTag.getAttribute(`title`)).toEqual(
      `Title for the image`
    )
    // No Intersection Observer in JSDOM, so placeholder img will be visible (opacity 1) by default
    expect(placeholderImageTag.getAttribute(`style`)).toEqual(
      `position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms; color: red;`
    )
    expect(placeholderImageTag.getAttribute(`class`)).toEqual(`placeholder`)
  })

  it(`should have a transition-delay of 1sec`, () => {
    const component = setup(false, { durationFadeIn: 1000 })
    expect(component).toMatchSnapshot()
  })

  it(`should have the "critical" prop set "loading='eager'"`, () => {
    jest.spyOn(global.console, `log`)

    const props = { critical: true }
    const imageTag = setup(false, props).querySelector(`picture img`)
    expect(imageTag.getAttribute(`loading`)).toEqual(`eager`)
    expect(console.log).toBeCalled()
  })

  it(`should warn if image variants provided are missing media keys.`, () => {
    jest.spyOn(global.console, `warn`)

    render(
      <Image
        backgroundColor
        className={`fixedImage`}
        style={{ display: `inline` }}
        title={`Title for the image`}
        alt={`Alt text for the image`}
        crossOrigin={`anonymous`}
        fluid={fluidImagesShapeMock.concat({
          aspectRatio: 2,
          src: `test_image_3.jpg`,
          srcSet: `some other srcSet`,
          srcSetWebp: `some other srcSetWebp`,
          sizes: `(max-width: 600px) 100vw, 600px`,
          base64: `string_of_base64`,
        })}
        itemProp={`item-prop-for-the-image`}
        placeholderStyle={{ color: `red` }}
        placeholderClassName={`placeholder`}
      />
    )
    expect(console.warn).toBeCalled()
  })

  it(`should warn if missing both fixed and fluid props`, () => {
    jest.spyOn(global.console, `error`)

    render(<Image fixed={null} />)

    expect(console.error).toBeCalledWith(
      expect.stringContaining(
        `The prop \`fluid\` or \`fixed\` is marked as required`
      )
    )
  })

  it(`should select the correct mocked image of fluid variants provided.`, () => {
    const mediaQuery = `only screen and (min-width: 1024px)`
    matchMedia.useMediaQuery(mediaQuery)

    const tripleFluidImageShapeMock = fluidImagesShapeMock.concat({
      aspectRatio: 5,
      src: `test_image_4.jpg`,
      srcSet: `third other srcSet`,
      srcSetWebp: `third other srcSetWebp`,
      sizes: `(max-width: 1920px) 100vw, 1920px`,
      base64: `string_of_base64`,
      media: `only screen and (min-width: 1024px)`,
    })
    const { container } = render(
      <Image
        backgroundColor
        className={`fluidArtDirectedImage`}
        style={{ display: `inline` }}
        title={`Title for the image`}
        alt={`Alt text for the image`}
        crossOrigin={`anonymous`}
        fluid={tripleFluidImageShapeMock}
        itemProp={`item-prop-for-the-image`}
        placeholderStyle={{ color: `red` }}
        placeholderClassName={`placeholder`}
      />
    )
    const aspectPreserver = container.querySelector(`div div div`)
    expect(aspectPreserver.getAttribute(`style`)).toEqual(
      expect.stringMatching(/padding-bottom: 20%/)
    )
  })

  it(`should select the correct mocked image of fixed variants provided.`, () => {
    const mediaQuery = `only screen and (min-width: 1024px)`
    matchMedia.useMediaQuery(mediaQuery)

    const tripleFixedImageShapeMock = fixedImagesShapeMock.concat({
      width: 1024,
      height: 768,
      src: `test_image_4.jpg`,
      srcSet: `third other srcSet`,
      srcSetWebp: `third other srcSetWebp`,
      base64: `string_of_base64`,
      media: `only screen and (min-width: 1024px)`,
    })
    const { container } = render(
      <Image
        backgroundColor
        className={`fixedArtDirectedImage`}
        style={{ display: `inline` }}
        title={`Title for the image`}
        alt={`Alt text for the image`}
        crossOrigin={`anonymous`}
        fixed={tripleFixedImageShapeMock}
        itemProp={`item-prop-for-the-image`}
        placeholderStyle={{ color: `red` }}
        placeholderClassName={`placeholder`}
      />
    )
    const aspectPreserver = container.querySelector(`div div`)
    expect(aspectPreserver.getAttribute(`style`)).toEqual(
      expect.stringMatching(/width: 1024px; height: 768px;/)
    )
  })

  it(`should select the image with no media query as mocked image of fluid variants provided.`, () => {
    const { container } = render(
      <Image
        backgroundColor
        className={`fluidArtDirectedImage`}
        style={{ display: `inline` }}
        title={`Title for the image`}
        alt={`Alt text for the image`}
        crossOrigin={`anonymous`}
        fluid={fluidImagesShapeMock.slice().reverse()}
        itemProp={`item-prop-for-the-image`}
        placeholderStyle={{ color: `red` }}
        placeholderClassName={`placeholder`}
      />
    )
    const aspectPreserver = container.querySelector(`div div div`)
    expect(aspectPreserver.getAttribute(`style`)).toEqual(
      expect.stringMatching(/padding-bottom: 50%/)
    )
  })

  it(`should select the image with no media query as mocked image of fixed variants provided.`, () => {
    const { container } = render(
      <Image
        backgroundColor
        className={`fixedArtDirectedImage`}
        style={{ display: `inline` }}
        title={`Title for the image`}
        alt={`Alt text for the image`}
        crossOrigin={`anonymous`}
        fixed={fixedImagesShapeMock.slice().reverse()}
        itemProp={`item-prop-for-the-image`}
        placeholderStyle={{ color: `red` }}
        placeholderClassName={`placeholder`}
      />
    )
    const aspectPreserver = container.querySelector(`div div`)
    expect(aspectPreserver.getAttribute(`style`)).toEqual(
      expect.stringMatching(/width: 100px; height: 100px;/)
    )
  })

  it(`should call onLoad and onError image events`, () => {
    const onLoadMock = jest.fn()
    const onErrorMock = jest.fn()
    const imageTag = setup(true, {}, onLoadMock, onErrorMock).querySelector(
      `picture img`
    )
    fireEvent.load(imageTag)
    fireEvent.error(imageTag)

    expect(onLoadMock).toHaveBeenCalledTimes(1)
    expect(onErrorMock).toHaveBeenCalledTimes(1)
  })

  it(`should have an "aria-hidden" attribute on the fluid element`, () => {
    const divTag = setup(true).querySelector(`.gatsby-image-wrapper > div`)
    expect(divTag.getAttribute(`aria-hidden`)).toBe(`true`)
  })

  it(`should have an "aria-hidden" attribute on the background element`, () => {
    const BgTag = setup(true).querySelector(`.gatsby-image-wrapper > div + div`)
    expect(BgTag.getAttribute(`aria-hidden`)).toBe(`true`)
  })

  it(`should have an "aria-hidden" attribute on the Placeholder component when fluid`, () => {
    const placeholderImageTag = setup(true).querySelector(`img`)
    expect(placeholderImageTag.getAttribute(`aria-hidden`)).toBe(`true`)
  })

  it(`should have an "aria-hidden" attribute on the Placeholder component when fixed`, () => {
    const placeholderImageTag = setup().querySelector(`img`)
    expect(placeholderImageTag.getAttribute(`aria-hidden`)).toBe(`true`)
  })

  it(`should not have an "aria-hidden" attribute on the Image`, () => {
    const placeholderImageTag = setup().querySelector(`picture img`)
    expect(placeholderImageTag.getAttribute(`aria-hidden`)).toBe(null)
  })
})
