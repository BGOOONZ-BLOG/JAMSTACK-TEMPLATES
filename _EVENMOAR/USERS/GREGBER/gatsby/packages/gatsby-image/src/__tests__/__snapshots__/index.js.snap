// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should have a transition-delay of 1sec 1`] = `
<div>
  <div
    class="fixedImage gatsby-image-wrapper"
    style="position: relative; overflow: hidden; display: inline; width: 100px; height: 100px;"
  >
    <div
      aria-hidden="true"
      style="background-color: lightgray; width: 100px; opacity: 1; height: 100px; transition-delay: 1000ms;"
      title="Title for the image"
    />
    <img
      alt=""
      aria-hidden="true"
      class="placeholder"
      itemprop="item-prop-for-the-image"
      src="string_of_base64"
      style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 1000ms; color: red;"
      title="Title for the image"
    />
    <picture>
      <source
        srcset="some srcSetWebp"
        type="image/webp"
      />
      <source
        srcset="some srcSet"
      />
      <img
        alt="Alt text for the image"
        crossorigin="anonymous"
        draggable="true"
        height="100"
        itemprop="item-prop-for-the-image"
        loading="lazy"
        src="test_image.jpg"
        srcset="some srcSet"
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 1000ms;"
        title="Title for the image"
        width="100"
      />
    </picture>
    <noscript>
      &lt;picture&gt;&lt;source type='image/webp' srcset="some srcSetWebp" /&gt;&lt;source srcset="some srcSet" /&gt;&lt;img loading="lazy" width="100" height="100" srcset="some srcSet" src="test_image.jpg" alt="Alt text for the image" title="Title for the image" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
  </div>
</div>
`;

exports[`<Image /> should render fixed size images 1`] = `
<div>
  <div
    class="fixedImage gatsby-image-wrapper"
    style="position: relative; overflow: hidden; display: inline; width: 100px; height: 100px;"
  >
    <div
      aria-hidden="true"
      style="background-color: lightgray; width: 100px; opacity: 1; height: 100px; transition-delay: 500ms;"
      title="Title for the image"
    />
    <img
      alt=""
      aria-hidden="true"
      class="placeholder"
      itemprop="item-prop-for-the-image"
      src="string_of_base64"
      style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms; color: red;"
      title="Title for the image"
    />
    <picture>
      <source
        srcset="some srcSetWebp"
        type="image/webp"
      />
      <source
        srcset="some srcSet"
      />
      <img
        alt="Alt text for the image"
        crossorigin="anonymous"
        draggable="true"
        height="100"
        itemprop="item-prop-for-the-image"
        loading="lazy"
        src="test_image.jpg"
        srcset="some srcSet"
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
        title="Title for the image"
        width="100"
      />
    </picture>
    <noscript>
      &lt;picture&gt;&lt;source type='image/webp' srcset="some srcSetWebp" /&gt;&lt;source srcset="some srcSet" /&gt;&lt;img loading="lazy" width="100" height="100" srcset="some srcSet" src="test_image.jpg" alt="Alt text for the image" title="Title for the image" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
  </div>
</div>
`;

exports[`<Image /> should render fluid images 1`] = `
<div>
  <div
    class="fixedImage gatsby-image-wrapper"
    style="position: relative; overflow: hidden; display: inline;"
  >
    <div
      aria-hidden="true"
      style="width: 100%; padding-bottom: 66.66666666666667%;"
    />
    <div
      aria-hidden="true"
      style="background-color: lightgray; position: absolute; top: 0px; bottom: 0px; opacity: 1; right: 0px; left: 0px; transition-delay: 500ms;"
      title="Title for the image"
    />
    <img
      alt=""
      aria-hidden="true"
      class="placeholder"
      itemprop="item-prop-for-the-image"
      src="string_of_base64"
      style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms; color: red;"
      title="Title for the image"
    />
    <picture>
      <source
        sizes="(max-width: 600px) 100vw, 600px"
        srcset="some srcSetWebp"
        type="image/webp"
      />
      <source
        sizes="(max-width: 600px) 100vw, 600px"
        srcset="some srcSet"
      />
      <img
        alt="Alt text for the image"
        crossorigin="anonymous"
        draggable="true"
        itemprop="item-prop-for-the-image"
        loading="lazy"
        sizes="(max-width: 600px) 100vw, 600px"
        src="test_image.jpg"
        srcset="some srcSet"
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
        title="Title for the image"
      />
    </picture>
    <noscript>
      &lt;picture&gt;&lt;source type='image/webp' srcset="some srcSetWebp" sizes="(max-width: 600px) 100vw, 600px" /&gt;&lt;source srcset="some srcSet" sizes="(max-width: 600px) 100vw, 600px" /&gt;&lt;img loading="lazy" sizes="(max-width: 600px) 100vw, 600px" srcset="some srcSet" src="test_image.jpg" alt="Alt text for the image" title="Title for the image" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
  </div>
</div>
`;

exports[`<Image /> should render multiple fixed image variants 1`] = `
<div>
  <div
    class="fixedImage gatsby-image-wrapper"
    style="position: relative; overflow: hidden; display: inline; width: 100px; height: 100px;"
  >
    <div
      aria-hidden="true"
      style="background-color: lightgray; width: 100px; opacity: 1; height: 100px; transition-delay: 500ms;"
      title="Title for the image"
    />
    <picture>
      <source
        media="only screen and (min-width: 768px)"
        srcset="other_string_of_base64"
      />
      <source
        srcset="string_of_base64"
      />
      <img
        alt=""
        aria-hidden="true"
        class="placeholder"
        itemprop="item-prop-for-the-image"
        src="string_of_base64"
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms; color: red;"
        title="Title for the image"
      />
    </picture>
    <picture>
      <source
        media="only screen and (min-width: 768px)"
        srcset="some other srcSetWebp"
        type="image/webp"
      />
      <source
        media="only screen and (min-width: 768px)"
        srcset="some other srcSet"
      />
      <source
        srcset="some srcSetWebp"
        type="image/webp"
      />
      <source
        srcset="some srcSet"
      />
      <img
        alt="Alt text for the image"
        crossorigin="anonymous"
        height="100"
        itemprop="item-prop-for-the-image"
        loading="lazy"
        src="test_image.jpg"
        srcset="some srcSet"
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
        title="Title for the image"
        width="100"
      />
    </picture>
    <noscript>
      &lt;picture&gt;&lt;source type='image/webp' media="only screen and (min-width: 768px)" srcset="some other srcSetWebp" /&gt;&lt;source media="only screen and (min-width: 768px)" srcset="some other srcSet" /&gt;&lt;source type='image/webp' srcset="some srcSetWebp" /&gt;&lt;source srcset="some srcSet" /&gt;&lt;img loading="lazy" width="100" height="100" srcset="some srcSet" src="test_image.jpg" alt="Alt text for the image" title="Title for the image" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
  </div>
</div>
`;

exports[`<Image /> should render multiple fluid image variants 1`] = `
<div>
  <div
    class="fixedImage gatsby-image-wrapper"
    style="position: relative; overflow: hidden; display: inline;"
  >
    <div
      aria-hidden="true"
      style="width: 100%; padding-bottom: 50%;"
    />
    <div
      aria-hidden="true"
      style="background-color: lightgray; position: absolute; top: 0px; bottom: 0px; opacity: 1; right: 0px; left: 0px; transition-delay: 500ms;"
      title="Title for the image"
    />
    <picture>
      <source
        media="only screen and (min-width: 768px)"
        srcset="string_of_base64"
      />
      <source
        srcset="string_of_base64"
      />
      <img
        alt=""
        aria-hidden="true"
        class="placeholder"
        itemprop="item-prop-for-the-image"
        src="string_of_base64"
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms; color: red;"
        title="Title for the image"
      />
    </picture>
    <picture>
      <source
        media="only screen and (min-width: 768px)"
        sizes="(max-width: 600px) 100vw, 600px"
        srcset="some other srcSetWebp"
        type="image/webp"
      />
      <source
        media="only screen and (min-width: 768px)"
        sizes="(max-width: 600px) 100vw, 600px"
        srcset="some other srcSet"
      />
      <source
        sizes="(max-width: 600px) 100vw, 600px"
        srcset="some srcSetWebp"
        type="image/webp"
      />
      <source
        sizes="(max-width: 600px) 100vw, 600px"
        srcset="some srcSet"
      />
      <img
        alt="Alt text for the image"
        crossorigin="anonymous"
        itemprop="item-prop-for-the-image"
        loading="lazy"
        sizes="(max-width: 600px) 100vw, 600px"
        src="test_image.jpg"
        srcset="some srcSet"
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
        title="Title for the image"
      />
    </picture>
    <noscript>
      &lt;picture&gt;&lt;source type='image/webp' media="only screen and (min-width: 768px)" srcset="some other srcSetWebp" sizes="(max-width: 600px) 100vw, 600px" /&gt;&lt;source media="only screen and (min-width: 768px)" srcset="some other srcSet" sizes="(max-width: 600px) 100vw, 600px" /&gt;&lt;source type='image/webp' srcset="some srcSetWebp" sizes="(max-width: 600px) 100vw, 600px" /&gt;&lt;source srcset="some srcSet" sizes="(max-width: 600px) 100vw, 600px" /&gt;&lt;img loading="lazy" sizes="(max-width: 600px) 100vw, 600px" srcset="some srcSet" src="test_image.jpg" alt="Alt text for the image" title="Title for the image" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
  </div>
</div>
`;
